nomenclature/processor/data_validator.py

```diff
@@ -91,30 +110,38 @@ def from_file(cls, file: Union[Path, str]) -> "DataValidator":
 
     def apply(self, df: IamDataFrame) -> IamDataFrame:
         error_list = []
+        warning_list = []
 
-        with adjust_log_level():
+        with adjust_log_level(level="WARNING"):
             for item in self.criteria_items:
                 failed_validation = df.validate(**item.validation_args)
                 if failed_validation is not None:
-                    error_list.append(
-                        "  Criteria: "
-                        + ", ".join(
-                            [f"{key}: {value}" for key, value in item.criteria.items()]
-                        )
+                    criteria_msg = "  Criteria: " + ", ".join(
+                        [f"{key}: {value}" for key, value in item.criteria.items()]
                     )
-                    error_list.append(
+                    if item.warning_level:
+                        log_list = warning_list
```

Comment:
As just discussed bilaterally, I find this implementation confusing and it fails to produce the expected result as soon as the validation file contains both warning and error constraints.